### PR TITLE
feat: PC-11815 add budget adjustments terraform support

### PR DESCRIPTION
**Pull request by kubaceg**
## Motivation

Add support for the new `BudgetAdjustment` kind.

## Release Notes

New BudgetAdjustment kind which allows customers to schedule recurring future events that should be excluded from the Error Budget calculations in their SLOs. This PR introduces new `budget_adjustment` resource.

### CODE REVIEW

File: nobl9/resource_budgetadjustment.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+			Description: "The time of the first event start. " +
+				"The expected value is a string with date in RFC3339 format. " +
+				"Example: `2022-12-31T00:00:00Z`",
+		},
+		"duration": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validateDuration,
+			Description: "The duration of the budget adjustment event. " +
+				"The expected value is a string in time duration string format. " +
```
Comment (labtom) [inline nobl9/resource_budgetadjustment.go:51]:
```suggestion
				"The expected value for this field is a string formatted as a time duration. " +
```

File: nobl9/resource_budgetadjustment.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+				"Duration must be defined with 1 minute precision." +
```
Comment (labtom) [inline nobl9/resource_budgetadjustment.go:52]:
```suggestion
				"The duration must be defined with a precision of 1 minute." +
```

**Review by labtom — CHANGES_REQUESTED** [review]

**Review by labtom — DISMISSED** [review]
JZK

File: nobl9/resource_budgetadjustment.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+}
+
+func resourceBudgetAdjustmentApply(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(ProviderConfig)
+	client, ds := getClient(config)
+	if ds != nil {
+		return ds
+	}
+
+	budgetAdjustment := marshalBudgetAdjustment(d)
```
Comment (nieomylnieja) [inline nobl9/resource_budgetadjustment.go:138]: shadows budgetAdjustment() function, maybe just (or `bg`):

```suggestion
	adjustment := marshalBudgetAdjustment(d)
```

File: nobl9/resource_budgetadjustment.go [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+	if len(objects) != 1 {
+		d.SetId("")
+		return nil
+	}
+	object := objects[0]
+	var diags diag.Diagnostics
+	var err error
+
+	err = d.Set("name", object.Metadata.Name)
+	diags = appendError(diags, err)
```
Comment (nieomylnieja) [inline nobl9/resource_budgetadjustment.go:224]: wouldn't it be simpler to simply write:

```suggestion
	diags = appendError(diags, d.Set("name", object.Metadata.Name))
```

**Review by nieomylnieja — CHANGES_REQUESTED** [review]

**Review by nieomylnieja — APPROVED** [review]
looks good!

**Pull request merged**